lib/systems/default.nix

```diff
@@ -83,6 +83,48 @@ let
       # TODO: deprecate args.rustc in favour of args.rust after 23.05 is EOL.
       rust = args.rust or args.rustc or { };
 
+      toolchain = {
+        cc =
+          if final.useLLVM || final.isDarwin then
+            "clang"
+          else if final.useArocc then
+            "arocc"
+          else if final.useZig then
+            "zig"
+          else
+            # TODO: don't imply GCC when nothing else is specified.
+            # Return null once we allow these attributes to be changed.
+            "gcc";
+
+        bintools =
+          if final.useLLVM || final.useArocc || final.useZig || final.isDarwin then "llvm" else "gnu";
```

Comment:
@reckenrode What do we want for Darwin here? I’m not sure what the balance of LLVM vs. `cctools` is currently.

Maybe it would be best to split this up into separate variables, but I’m not sure what the correct split would be.

Reply:
Darwin’s bintools is almost all LLVM except for the following: `ranlib`, `otool`, `install_name_tool`, and `ld`. I’m currently testing a change to use `ranlib` from LLVM, so it might only be the last three for 25.11.

I’d really like to change the linker to `ld64` to match what Darwin is actually using. Technically, cctools has another linker, but it’s _old_ and not used anymore.

Reply:
(In short, I think `llvm` is right for Darwin for `bintools` here.)